src/Validator/DataSurveyAnswerTypeChoiceValidator.php

```diff
@@ -0,0 +1,81 @@
+<?php
+
+namespace AppBundle\Validator;
+
+use AppBundle\Entity\Jecoute\DataAnswer;
+use AppBundle\Jecoute\SurveyQuestionTypeEnum;
+use Symfony\Component\Validator\Constraint;
+use Symfony\Component\Validator\ConstraintValidator;
+use Symfony\Component\Validator\Exception\UnexpectedTypeException;
+
+class DataSurveyAnswerTypeChoiceValidator extends ConstraintValidator
+{
+    /**
+     * @param DataAnswer $answer
+     */
+    public function validate($answer, Constraint $constraint)
+    {
+        if (!$constraint instanceof DataSurveyAnswerTypeChoice) {
+            throw new UnexpectedTypeException($constraint, DataSurveyAnswerTypeChoice::class);
+        }
+
+        if (null === $answer || null === $answer->getSurveyQuestion()) {
+            return;
+        }
+
+        $surveyQuestion = $answer->getSurveyQuestion();
+
+        switch ($surveyQuestion->getQuestion()->getType()) {
+            case SurveyQuestionTypeEnum::SIMPLE_FIELD:
+                if (!$answer->getSelectedChoices()->isEmpty()) {
+                    $this->processViolation($constraint->simpleFieldAnswerMessage, $surveyQuestion->getId());
+                }
+
+                if (!$answer->getTextField()) {
+                    $this->processViolation(
+                        $constraint->simpleFieldAnswerWithoutTextFieldMessage,
+                        $surveyQuestion->getId()
+                    );
+                }
+
+                break;
+            case SurveyQuestionTypeEnum::UNIQUE_CHOICE_TYPE:
+                if (1 !== $answer->getSelectedChoices()->count()) {
+                    $this->processViolation($constraint->uniqueChoiceAnswerMessage, $surveyQuestion->getId());
+                }
+
+                if ($answer->getTextField()) {
+                    $this->processViolation(
+                        $constraint->uniqueChoiceAnswerWithTextFieldMessage,
+                        $surveyQuestion->getId()
+                    );
+                }
+
+                break;
+            case SurveyQuestionTypeEnum::MULTIPLE_CHOICE_TYPE:
+                if ($answer->getSelectedChoices()->count() <= 1) {
```

Comment:
you mean isEmpty() ? 

i would allow 1 choice selected in a "multiple choice questinos"